modules/rpc/yarpc.go

```diff
@@ -105,78 +196,85 @@ func newYarpcModule(
 		}
 	}
 
-	err := module.Host().Config().Get(fmt.Sprintf("modules.%s", module.Name())).PopulateStruct(cfg)
-	// found values, update module
-	module.config = *cfg
+	err = module.Host().Config().Scope("modules").Get(module.Name()).PopulateStruct(&module.config)
+	if err != nil {
+		return module, err
+	}
+
+	module.config.inboundMiddleware = inboundMiddlewareFromCreateInfo(mi)
+	module.config.onewayInboundMiddleware = onewayInboundMiddlewareFromCreateInfo(mi)
 
-	module.inboundMiddlewares = inboundMiddlewaresFromCreateInfo(mi)
-	module.onewayInboundMiddlewares = onewayInboundMiddlewaresFromCreateInfo(mi)
+	tchTransport, err := tch.NewChannelTransport(
+		tch.ServiceName(module.config.AdvertiseName),
+		tch.ListenAddr(module.config.Bind),
+	)
+
+	module.config.inbounds = []transport.Inbound{tchTransport.NewInbound()}
 
 	return module, err
 }
 
-// Start begins serving requests over YARPC
-func (m *YarpcModule) Start(readyCh chan<- struct{}) <-chan error {
+// Start begins serving requests over RPC. It first stops the current dispatcher, merges configs from the global
+// collection and then starts the new dispatcher.
+func (m *YARPCModule) Start(readyCh chan<- struct{}) <-chan error {
```

Comment:
Start/Stop should only be called once we've defined all the Modules of the application right?  Do we see a situation where after Start is called on a YARPCModule we'd want to add another YARPCModule?  Or is the plan:

1) Configure/Create all Modules
2) Start all Modules
3) Let application run
4) Get shutdown signal, call stop on all Modules

With no dynamic module creation.